packages/react/src/auth/useFetchSignInMethodsForEmailQuery.ts

```diff
@@ -0,0 +1,23 @@
+import { type UseQueryOptions, useQuery } from "@tanstack/react-query";
+import {
+  fetchSignInMethodsForEmail,
+  type Auth,
+  type AuthError,
+} from "firebase/auth";
+
+type AuthUseQueryOptions<
+  TData = unknown,
+  TError = Error,
+  TVariables = void
+> = Omit<UseQueryOptions<TData, TError, TVariables>, "queryFn">;
+
+export function useFetchSignInMethodsForEmailQuery(
+  auth: Auth,
+  email: string,
+  options: AuthUseQueryOptions<string[], AuthError, void>
+) {
+  return useQuery<string[], AuthError, void>({
+    ...options,
+    queryFn: () => fetchSignInMethodsForEmail(auth, email),
```

Comment:
The email should be passed as a mutation value

Reply:
Would this have change to be a mutation? 

Currently, this is a query versus mutations for which a user can call a `mutate()` or a `mutateAsync` method.